main.cc

```diff
@@ -170,6 +171,54 @@ read_config(bpo::variables_map& opts, db::config& cfg) {
     });
 }
 
+struct object_storage_endpoint_param {
+    sstring endpoint;
+    s3::endpoint_config config;
+};
+
+namespace YAML {
+template<>
+struct convert<::object_storage_endpoint_param> {
+    static bool decode(const Node& node, ::object_storage_endpoint_param& ep) {
+        ep.endpoint = node["name"].as<std::string>();
+        return true;
+    }
+};
+}
+
+// run inside async context
+static std::unordered_map<sstring, s3::endpoint_config> read_object_storage_config(const db::config& cfg) {
+    std::unordered_map<sstring, s3::endpoint_config> ret;
+    sstring file;
+    if (!cfg.object_storage_config_file().empty()) {
+        file = cfg.object_storage_config_file();
+    } else {
+        file = db::config::get_conf_sub("object_storage.yaml").native();
+    }
+    YAML::Node doc;
+    try {
+        doc = YAML::LoadFile(file);
+    } catch (YAML::BadFile&) {
+        // Missing file -- OK, just return empty config
+        startlog.info("No {} file found, skipping", file);
+        return ret;
+    }
+
+    for (auto&& section : doc) {
+        auto sec_name = section.first.as<std::string>();
+        if (sec_name != "endpoints") {
+            throw std::runtime_error(fmt::format("While parsing object_storage config: section {} currently unsupported.", sec_name));
+        }
+
+        auto endpoints = section.second.as<std::vector<object_storage_endpoint_param>>();
+        for (auto&& ep : endpoints) {
+            ret[ep.endpoint] = std::move(ep.config);
+        }
+    }
```

Comment:
The format of this file should be documented somewhere.

Reply:
Added `docs/dev/object_storage.md`